core/lib/tee_verifier/src/lib.rs

```diff
@@ -0,0 +1,714 @@
+//! Tee verifier
+//!
+//! Verifies that a L1Batch has the expected root hash after
+//! executing the VM and verifying all the accessed memory slots by their
+//! merkle path.
+//!
+//! The crate/library shadows some structures to avoid polluting the global code
+//! base with a stable (De-)Serialize ABI.
+//! If additional components need to (De-)Serialize these structures, we can revisit this decision.
+//!
+//! This crate can't be put in `zksync_types`, because it would add a circular dependency.
```

Comment:
I don't think it should, honestly (`zksync_types` is too big already and has to be split), so you may remove (or leave :) ) this comment.

Reply:
removed